src/main/java/com/cloudant/client/internal/URIBaseMethods.java

```diff
@@ -0,0 +1,130 @@
+package com.cloudant.client.internal;
+
+import com.cloudant.client.org.lightcouch.Params;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+
+/**
+ * Provides methods using fluent style setters to build a Cloudant account or database URI.
+ * The use of generics provides the subclasses a way to re-use common URI methods with
+ * different types. Current types are {@link DatabaseURIHelper} for creating a URI to
+ * interact with a Cloudant database, and {@link URIBase} for interacting with a
+ * Cloudant account.
+ */
+abstract class URIBaseMethods<T extends URIBaseMethods> {
+
+    private static final String _design_prefix_encoded = "_design%2F";
+    private static final String _local_prefix_encoded = "_local%2F";
+    /* key=value params */
+    Params qParams = new Params();
+    String completeQuery = "";
+    String path = "";
+    URI baseUri;
+
+    public abstract T returnThis();
+
+    /**
+     * Sets a path(s) for the URI.
+     * Note: File separator is handled in this method
+     * and should not be passed as parameter.
+     *
+     * @return The updated {@link T} object.
+     */
+    public T path(String path) {
+        this.path += "/" + encodePath(path);
```

Comment:
I would honestly prefer to avoid either situation occurring.

How about
```
if (path.length() == 0) {
  this.path += encodePath(path);
} else {
  this.path += "/" + encodePath(path);
}
```
?